micro-rdk/src/common/data_collector.rs

```diff
@@ -0,0 +1,352 @@
+        //     params
+        // } else {
+        //     &Kind::NullValue(0)
+        // };
+        let method = match method_str.as_str() {
+            "Readings" => CollectionMethod::Readings,
+            "AngularVelocity" => CollectionMethod::AngularVelocity,
+            "LinearAcceleration" => CollectionMethod::LinearAcceleration,
+            "LinearVelocity" => CollectionMethod::LinearVelocity,
+            // TODO: Power Sensor and Board collectors
+            // "Voltage" => CollectionMethod::Voltage,
+            // "Current" => CollectionMethod::Current,
+            // "Analogs" => {
+            //     let analog_reader_name: String = parameters.get("reader_name")?.ok_or(AttributeError::KeyNotFound("reader_name".to_string()))?.try_into()?;
+            //     CollectionMethod::Analogs(analog_reader_name)
+            // },
+            _ => {
+                return Err(AttributeError::ConversionImpossibleError);
+            }
+        };
+        Ok(DataCollectorConfig {
+            method,
+            capture_frequency_hz,
+        })
+    }
+}
+
+/// A CollectionMethod is an enum whose values are associated with
+/// a method on one or more component traits
+#[derive(Debug, Clone)]
+pub enum CollectionMethod {
+    Readings,
+    // MovementSensor methods
+    AngularVelocity,
+    LinearAcceleration,
+    LinearVelocity,
+    // TODO: PowerSensor methods
+    // Voltage,
+    // Current,
+    // TODO: Board
+    // Analogs(String)
+}
+
+impl CollectionMethod {
+    fn method_str(&self) -> String {
+        match self {
+            Self::Readings => "readings".to_string(),
+            Self::AngularVelocity => "angularvelocity".to_string(),
+            Self::LinearAcceleration => "linearacceleration".to_string(),
+            Self::LinearVelocity => "linearvelocity".to_string(),
+            // TODO: Power Sensor and Board collectors
+            // Self::Voltage => "voltage".to_string(),
+            // Self::Current => "current".to_string(),
+            // Self::Analogs(_) => "analogs".to_string()
+        }
+    }
+}
+
+impl Display for CollectionMethod {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        std::fmt::Display::fmt(&self.method_str(), f)
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum DataCollectionError {
+    #[error("method {0} unsupported for {1}")]
+    UnsupportedMethod(CollectionMethod, String),
+    #[error("no collection methods supported for component")]
+    NoSupportedMethods,
+    #[error(transparent)]
+    SensorCollectionError(#[from] SensorError),
+    // TODO: remove when error enums for all components are created (RSDK-6909)
+    #[error(transparent)]
+    MiscCollectionError(#[from] anyhow::Error),
+}
+
+/// A DataCollector represents an association between a data collection method and
+/// a ResourceType (i.e. SensorType & Readings, BoardType & Analogs) and the frequency at
+/// which the results of the method should be stored. It is capable
```

Comment:
Looks like the last sentence is incomplete. I was going to say a sentence fragement, but that isn't exactly true.

Reply:
yeah, should have backspaced harder